pkg/volume/glusterfs/glusterfs.go

```diff
@@ -771,6 +771,21 @@ func (p *glusterfsVolumeProvisioner) CreateVolume(gid int) (r *v1.GlusterfsVolum
 	}
 	glog.V(2).Infof("create volume of size %dGiB", sz)
 
+	// Check whether we could create an endpoint before we
+	// create the gluster volume in order to prevent the churn
+	// of creating and deleting the volume if we already know
+	// the overall operation will fail anyways.
+	epServiceName := dynamicEpSvcPrefix + p.options.PVC.Name
```

Comment:
My suggestion is to do the PVC namelength as start of the provision() , reason being this has an unwanted gidallocation when we are sure that, its going to fail. Secondly, the `63` characters check is already imposed by the API server any way.

Reply:
That's a good point!

In order to that right, I'd rather refactor the code a little more, to take the endpoint creation out of the CreateVolume() function, where it imo does not belong anyways. Will update the PR once done.

Reply:
but, why we cant do len() check on pvc name  instead of sticking on service name @obnoxxx ?

Reply:
Because that's not a robust way to do this: The limitation is not actually on the PVC name, but on the resulting endpoint name. So if you check the PVC name length, and at some point change the way an endpoint name is generated, then you need to change this check too. (==> bad.) But if you generate the name first and check for the resulting endpoint name, then the check stays unaffected of possible future changes of how to create the endpoint name. (==> good!)

That's why I want to do the check this way.